documentation/src/main/asciidoc/reference/limitations.asciidoc

```diff
@@ -0,0 +1,79 @@
+[[limitations]]
+= Limitations
+// Search 5 anchors backward compatibility
+[[elasticsearch-limitations]]
+
+[[limitations-parallel-embedded-update]]
+== In rare cases, automatic indexing involving `@IndexedEmbedded` may lead to out-of sync indexes
+
+=== Description
+
+If two entity instances are <<mapper-orm-indexedembedded,indexed-embedded>> in the same "index-embedding" entity,
+and these two entity instance are updated in parallel transactions,
+there is a small risk that the transaction commits happen in just the wrong way,
+leading to the index-embedding entity being reindexed with only part of the updates.
+
+For example, consider indexed entity A, which index-embeds B and C.
+The following course of events involving two parallel transactions (T1 and T2)
+will lead to an out of date index:
+
+* T1: Load B.
+* T1: Change B in a way that will require reindexing A.
+* T2: Load C.
+* T2: Change C in a way that will require reindexing A.
+* T2: Request the transaction commit.
+  Hibernate Search builds the document for A.
+  While doing so, it automatically loads B. B appears unmodified, as T1 wasn't committed yet.
+* T1: Request the transaction commit.
+  Hibernate Search builds documents to index.
+  While doing so, it automatically loads C. C appears unmodified, as T2 wasn't committed yet.
+* T1: Transaction is committed.
+  Hibernate Search automatically sends the updated A to the index.
+  In this version, B is updated, but C is not.
+* T2: Transaction is committed.
+  Hibernate Search automatically sends the updated A to the index.
+  In this version, C is updated, but B is not.
+
+This chain of events ends with the index containing a version of A where C is updated, but B is not.
+
+=== Workaround
+
+The following solutions can help circumvent this limitation:
+
+1. Avoid parallel updates to entities that are indexed-embedded in the same indexed entity.
+This is only possible in very specific setups.
+2. OR schedule a <<mapper-orm-indexing-massindexer,full reindexing>> of your database periodically (e.g. every night)
+to get the index back in sync with the database.
```

Comment:
I would like to add a third option here (if it makes sense):
Remove a re-indexing trigger. E.g.: from C -> A.
But I'm aware that many to one and many to many are very common with a relational data store.

Reply:
While it would eliminate the problem, this solution would also lead to out-of-date indexes, which is what we want to avoid here. So I don't think it's worth mentioning.